probe_src/python/probe_py/manual/workflows.py

```diff
@@ -0,0 +1,267 @@
+from probe_py.manual.analysis import *
+import networkx as nx
+import dataclasses
+from dataclasses import dataclass
+from typing import Tuple, List, Set, Union
+
+"""
+All the cases we should take care of:
+1- One Input, One Output
+2- One Input, Multiple Outputs
+3- Multiple Inputs, One Output
+4- Multiple Inputs, Multiple Outputs
+5- Chained Commands: Inline commands that directly modify the input file (e.g., using sed, awk, or similar) 
+6- No Input Command: Commands like ls .: Commands that don't take an explicit input file but generate output 
+7- Ensure that any environment variables or context-specific settings are captured and translated into the Nextflow environment
+8- File and Directory Structure Assumptions (Scripts that assume a specific directory structure, Commands that change the working directory (cd))
+...
+"""
+
+class NextflowGenerator:
+    def __init__(self, graph: nx.DiGraph):
```

Comment:
I think `graph` should be an argument to `generate_workflow`, since inheritance works differently on `__init__` than on regular methods. `__init__` should be used for *generator-specific configuration options*. `generate_workflow` should be stateless.

We should  have an [abstract base class](https://docs.python.org/3/library/abc.html) that these generators derive from:

```python
import abc

class Generator(abc.ABC):
    @abc.abstractmethod
    def generate_workflow(self, graph: nx.DiGraph) -> str:
        ...
        # literally three dots, since this is the abstract base class.
```